### ISSUE TITLE
works with 24port model, too

### ISSUE THREAD

**Issue by ka7** [BookGin/hp1820-cli#4]
just want to confirm that this works with the 24port model too.  ( j9980a )
( at least some basic-testing )

**Comment by BookGin** [comment]
That's cool. Thanks for your confirmation.

By the way, this project is very obsolete. I'm trying to implement [Ansible playbook](http://docs.ansible.com/ansible/latest/user_guide/playbooks.html) to substitute this project.

**Comment by ypid** [comment]
@BookGin Did you get something working with Ansible?

**Comment by BookGin** [comment]
@ypid No, I no longer worked on this.

**Comment by ypid** [comment]
So this project is not yet obsolete/superseded? Because I did not quite understand how Ansible could just replace this project. Even Ansible needs modules which can talk to the switch.

**Comment by BookGin** [comment]
Yes, Ansible doesn't have similar modules as far as I know. My idea is to rewrite this project as an [Ansible custom module](https://docs.ansible.com/ansible/2.5/dev_guide/developing_modules.html), but not replace it with Ansible.

I think this project is still usable so it's not yet obsolete :smiley:

**Comment by ypid** [comment]
Got it. Then we have the same understanding :) I would be interesting in a module and help test it, but this is not something I need short-term. Of course there is the question if this switch is even suitable or if it would be easier to just get a proper switch (Cumulus Linux comes to mind, although maybe outside of what I would like to spend), but I guess with your ground work, I am kinda on board with just hacking together a solution for what is basically (intentional) software limitation.

**Comment by ypid** [comment]
I tested it now a bit. The emulated cli is pretty nice :+1: And I thought if I should write an Ansible module for desired state configuration for VLAN configuration from NetBox. I see two options:

1. Get a machine readable version of `printVlanMembership` and compare it to the desired state (passed as dict variable with all VLANs). Than issue `accessVlan` and other calls on the diff. This should be the fastest.
2. Generate the config somehow. I did not find details on the config format. Seems to be some binary one. I looked up a string I found in the hexdump `simcfgData.cfg` in the hope someone would have details and a encoder/decoder. Anyway, I guess this is not the way to go here because config upload causes a reload which takes down the switching engine for 10 seconds.

I would prefer 1. ~~It would take a few hours to do that and I have decided against doing it. Not really worth it for my one switch of this model :)~~ Turns out I want the consistency check from NetBox to the device after all, so I am working on it.

**Comment by ypid** [comment]
Done, see #5. VLAN is all I need for now, probably ever. I did not do any refactoring like rename of the module ("lib" is not an ideal name, I suggest "hpe1820") or similar.
